File: java/estruturas-repetitivas/Experiencias.java
import java.util.Locale;
import java.util.Scanner;

public class Experiencias {
    public static void main(String[] args) {
        Locale.setDefault(Locale.US);
        Scanner sc = new Scanner(System.in);
        
        int x, rato, sapo, coelho, total, i, num;
        char cobaia;
        double percentual_sapo, percentual_coelho, percentual_rato;

        System.out.print("Quantos casos de teste serao digitados?: ");
        x = sc.nextInt();

        rato = 0;
        sapo = 0;
        coelho = 0;

        for (i = 1; i <= x; i++) {
           System.out.print("Quantidade de cobaias: ");
           num = sc.nextInt();
           System.out.print("Tipo de cobaia: ");
           cobaia = sc.next().charAt(0);

           if (cobaia == 'R') {
            rato = rato + num;
           } else {
            if (cobaia == 'S') {
                sapo = sapo + num;
            } else {
                coelho = coelho + num;
            }
           }            
        }

        total = coelho + rato + sapo;
        percentual_coelho = (double) coelho / total * 100;
        percentual_rato = (double) rato / total * 100;
        percentual_sapo = (double) sapo / total * 100;

        System.out.println("RELATORIO FINAL: ");
        System.out.printf("Total: %d cobaias %n", total);
        System.out.printf("Total de coelhos: %d %n", coelho);
        System.out.printf("Total de ratos: %d %n", rato);
        System.out.printf("Total de sapos: %d %n", sapo);
        System.out.printf("Percentual de coelhos: %.2f %n", percentual_coelho);
        System.out.printf("Percentual de ratos: %.2f %n", percentual_rato);
        System.out.printf("Percentual de sapos: %.2f %n", percentual_sapo);


        sc.close();
    }
}

File: js/estruturas-repetitivas/experiencias.js
const readline = require('readline-sync');

const X = Number(readline.question("Quantos casos de teste serao digitados? "));
let coelho = 0;
let rato = 0;
let sapo = 0;

for (let i = 1; i <= X; i++) {
    let quant = Number(readline.question("Quantidade de cobaias: "));
    let cobaia = readline.question("Tipo de cobaia: ");

    if (cobaia === 'C') {
        coelho += quant;
    }

    if (cobaia === 'R') {
        rato += quant;
    }

    if (cobaia === 'S') {
        sapo += quant;
    }
}

const total = coelho + rato + sapo;
const percentualCoelho = (coelho / total) * 100;
const percentualRato = (rato / total) * 100;
const percentualSapo = (sapo / total) * 100;

console.log("RELATORIO FINAL:");
console.log(`Total: ${total} cobaias`);
console.log(`Total de coelhos: ${coelho}`);
console.log(`Total de ratos: ${rato}`);
console.log(`Total de sapos: ${sapo}`);
console.log(`Percentual de coelhos: ${percentualCoelho.toFixed(2)}`);
console.log(`Percentual de ratos: ${percentualRato.toFixed(2)}`);
console.log(`Percentual de sapos: ${percentualSapo.toFixed(2)}`);